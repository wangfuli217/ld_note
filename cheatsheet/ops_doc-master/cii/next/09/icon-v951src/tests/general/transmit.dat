Version 7 of Icon corrects a number of problems
with co-expressions that existed in previous
versions. The most significant one is proper
handling of co-expression return points.
