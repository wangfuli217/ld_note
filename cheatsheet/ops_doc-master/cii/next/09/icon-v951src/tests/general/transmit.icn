global words, lines, writer

procedure main()
   if not(&features == "co-expressions") then
      stop("co-expressions not supported")
   &trace := -1

   words := create word()
   lines := create reader()
   writer := create output()
   @writer
end

procedure word()
   static letters
   initial letters := &lcase ++ &ucase
   while line := @lines do
      line ? while tab(upto(letters)) do
         tab(many(letters)) @ writer
end

procedure reader()
   while read() @ words
end

procedure output()
   while write(&errout,@words)
   @&main
end
