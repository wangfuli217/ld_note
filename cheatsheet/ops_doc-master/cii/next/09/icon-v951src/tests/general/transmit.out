transmit.icn :   11  | main; co-expression_1 : &null @ co-expression_4
transmit.icn :   10  | output()
transmit.icn :   27  | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :    8  | | word()
transmit.icn :   17  | | | word; co-expression_2 : &null @ co-expression_3
transmit.icn :    9  | | | reader()
transmit.icn :   23  | | | | reader; co-expression_3 : "Version 7 of Ico..." @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "Version" @ co-expression_4
Version
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "of" @ co-expression_4
of
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "Icon" @ co-expression_4
Icon
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "corrects" @ co-expression_4
corrects
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "a" @ co-expression_4
a
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "number" @ co-expression_4
number
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "of" @ co-expression_4
of
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "problems" @ co-expression_4
problems
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   17  | | | | word; co-expression_2 : &null @ co-expression_3
transmit.icn :   23  | | | | reader; co-expression_3 : "with co-expressi..." @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "with" @ co-expression_4
with
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "co" @ co-expression_4
co
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "expressions" @ co-expression_4
expressions
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "that" @ co-expression_4
that
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "existed" @ co-expression_4
existed
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "in" @ co-expression_4
in
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "previous" @ co-expression_4
previous
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   17  | | | | word; co-expression_2 : &null @ co-expression_3
transmit.icn :   23  | | | | reader; co-expression_3 : "versions. The mo..." @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "versions" @ co-expression_4
versions
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "The" @ co-expression_4
The
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "most" @ co-expression_4
most
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "significant" @ co-expression_4
significant
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "one" @ co-expression_4
one
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "is" @ co-expression_4
is
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "proper" @ co-expression_4
proper
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   17  | | | | word; co-expression_2 : &null @ co-expression_3
transmit.icn :   23  | | | | reader; co-expression_3 : "handling of co-e..." @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "handling" @ co-expression_4
handling
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "of" @ co-expression_4
of
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "co" @ co-expression_4
co
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "expression" @ co-expression_4
expression
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "return" @ co-expression_4
return
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   19  | | | | word; co-expression_2 : "points" @ co-expression_4
points
transmit.icn :   27  | | | | output; co-expression_4 : &null @ co-expression_2
transmit.icn :   17  | | | | word; co-expression_2 : &null @ co-expression_3
transmit.icn :   24  | | | reader failed
transmit.icn :    9  | | | main; co-expression_3 failed to co-expression_2
transmit.icn :   20  | | word failed
transmit.icn :    8  | | main; co-expression_2 failed to co-expression_4
transmit.icn :   28  | | output; co-expression_4 : &null @ co-expression_1
transmit.icn :   12  | main failed
